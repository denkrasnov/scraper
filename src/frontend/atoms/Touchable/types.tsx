import { ReactNode } from "react";

import { DataAttr } from "~app/common/types";

export interface TouchableProps {
  children: ReactNode;
  dataAttr?: DataAttr;
  href?: string;
  productHover?: boolean;
  width?: string;
}
