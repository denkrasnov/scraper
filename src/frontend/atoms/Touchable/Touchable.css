.root {
  transition: transform 330ms ease-in-out;
}

.root:link {
  color: inherit;
  text-decoration: none;
}

.root:visited {
  color: inherit;
  text-decoration: none;
}

.root:hover {
  color: inherit;
  text-decoration: none;
}

.root:active {
  color: inherit;
  text-decoration: none;
}

@media (min-width: 992px) {
  .productHover:hover {
    transform: scale(1.04);
  }
}
