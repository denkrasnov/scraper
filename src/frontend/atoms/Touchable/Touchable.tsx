import { FC } from "react";

import { clearStyle } from "../helpers/clearStyle";
import mapPropsToStyles from "../helpers/mapPropsToStyles";
import { TouchableProps } from "./types";
import styles from "./Touchable.css";
import { StyleType } from "../types";
import Box from "../Box";

const Touchable: FC<TouchableProps> = (props) => {
  const { children, href, dataAttr, width, ...restProps } = props;
  const classes = mapPropsToStyles(restProps, styles);

  const style: StyleType = { width };

  return href ? (
    <a
      className={classes}
      href={href}
      rel="noopener noreferrer"
      style={clearStyle(style)}
      target="_blank"
      {...dataAttr}
    >
      {children}
    </a>
  ) : (
    <Box className={classes}>{children}</Box>
  );
};

export default Touchable;
