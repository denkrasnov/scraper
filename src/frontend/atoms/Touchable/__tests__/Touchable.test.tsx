import { shallow } from "enzyme";

import Touchable from "..";

describe("Touchable", () => {
  it("should render", () => {
    const component = shallow(<Touchable>Children</Touchable>);

    expect(component).toMatchSnapshot();
  });

  it("should render with href", () => {
    const component = shallow(<Touchable href="__URL__">Children</Touchable>);

    expect(component).toMatchSnapshot();
  });

  it("should render with width", () => {
    const component = shallow(<Touchable width="100%">Children</Touchable>);

    expect(component).toMatchSnapshot();
  });
});
