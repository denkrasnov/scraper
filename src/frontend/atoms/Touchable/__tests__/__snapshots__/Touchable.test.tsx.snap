// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Touchable should render 1`] = `
<a
  className="root"
  href="#"
  rel="noopener noreferrer"
  style={Object {}}
  target="_blank"
>
  Children
</a>
`;

exports[`Touchable should render with href 1`] = `
<a
  className="root"
  href="__URL__"
  rel="noopener noreferrer"
  style={Object {}}
  target="_blank"
>
  Children
</a>
`;

exports[`Touchable should render with width 1`] = `
<a
  className="root"
  href="#"
  rel="noopener noreferrer"
  style={
    Object {
      "width": "100%",
    }
  }
  target="_blank"
>
  Children
</a>
`;
